src/dbnode/client/write_state.go

```diff
@@ -33,6 +33,18 @@ import (
 	"github.com/m3db/m3/src/x/serialize"
 )
 
+type shardStatus int64
+
+const (
+	Undefined shardStatus = iota
```

Comment:
two separate nits: 
(1) these don't need to be exported. i.e. use `available`, not `Available`
(1) prefix enum values with enum type names. i.e. `shardStatusAvailable` instead of `available` (with the caveat that you should change update shardStatus as pointed above)